### PR TITLE
Fix for issue #2088 Night mode: Category pages

### LINKED ISSUE #2088: Night mode: Category pages
**Summary:** 

When clicking on a category in night mode, the app does not show the "Parent Categories", there's a blank space instead.

**Steps to reproduce:** 

How can we reproduce the issue? 
Settings > Night Mode;
Explore > Click any image > Click on a random categorie > Parent Categories

What did you expect the app to do, and what did you see instead?
The names of the parent categories should be shown.

**Device and Android version:** 

Found on Xiaomi A1, Android 8.1.0

**Commons app version:** 
2.8.5

**Screen-shots:** 
9a656a2e-8237-4dbe-b4b0-16f5b277ae89

**Would you like to work on the issue?**

Yess.

### PR DESCRIPTION
**Pull request by Divyanshu2132**
**Description (required)**

Fixes #2088 Night mode: Category pages

Checked for night mode in SearchCategoriesRenderer class and inflated layout accordingly

**Tests performed (required)**

Tested  Beta debug on Asus Zenfone Max Pro M1 with API level 27

**Screenshots showing what changed (optional - for UI changes)**
whatsapp image 2018-12-21 at 3 30 57 am

### CODE REVIEW

**Comment by domdomegg** [comment]
Screenshots of 318b0b1

screenshot_1545345847
screenshot_1545345854
screenshot_1545345860

**Comment by Divyanshu2132** [comment]
Sorry, I overlooked the tabs

**Comment by domdomegg** [comment]
Oh wait sorry for some reason I had in my mind that this was the fix for the tabs. Don't worry about them, they'll be done separately.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <uses-permission android:name="android.permission.READ_SYNC_STATS" />
     <uses-permission android:name="android.permission.WRITE_SYNC_SETTINGS" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
```
Comment (domdomegg) [inline app/src/main/AndroidManifest.xml]: Should not change this file

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java [hunk 2]
```diff
@@ -251,4 +251,4 @@ private void updateAllDatabases() {
     public interface LogoutListener {
         void onLogoutComplete();
     }
-}
+}
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Should not change this file

File: app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
             recent_searches_delete_button.setVisibility(View.GONE);
             recent_searches_text_view.setText(R.string.no_recent_searches);
         }
-  
+
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java]: Should not change this file

File: app/src/main/java/fr/free/nrw/commons/category/SubCategoryListFragment.java [hunk 4]
```diff
@@ -66,6 +67,11 @@
 
     });
 
+    @Override
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/category/SubCategoryListFragment.java]: Should not override this method

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Divyanshu2132) [reply app/src/main/AndroidManifest.xml]: Actually these were changes done previously on master branch of apps-android-commons

Comment (domdomegg) [reply app/src/main/AndroidManifest.xml]: Github only highlights the changes between your branch and master.

See https://github.com/commons-app/apps-android-commons/blob/78ca1fe64eff484723aa815a82b60bc33483ed3e/app/src/main/AndroidManifest.xml

**Comment by domdomegg** [comment]
Closing via #2213

**Pull request closed without merging**